app/models/commit.rb

```diff
@@ -83,6 +83,19 @@ def refresh_statuses!
     end
   end
 
+  def add_status(status_attributes)
+    previous_status = significant_status
+    statuses.create!(status_attributes)
+    reload
+    new_status = significant_status
+    if (previous_status != new_status) # we only care if something changed
+      if (!previous_status.success? && new_status.success?) || !new_status.success?
```

Comment:
I think `!new_status.pending?`. Should be enough.

"If the significant_status changed, and that it's not pending."